src/lib/detection_algorithm_manager/main.js

```diff
@@ -0,0 +1,42 @@
+//read from config to find detection algorthims and paths
+const path = require('path');
+const config_path = path.join(__dirname, 'config.json')
```

Comment:
Let's do the change that Limor described and get rid of the config.json file.
It should be quick -
Use fs.listdirSync (sync version is OK since this happen at require-time)  to iterate over the folder of detection algorithm.

Then, for each detection algorithm add the following to the exports at main .js:

exports.displayName (which will be like the display name at the JSON)
exports.hide (which will default to false, but you need to hide the "interface" folder)

Then change the code here to get this directly from the folder instead of the config.json